crates/symbolicator/src/endpoints/applecrashreport.rs

```diff
@@ -11,6 +15,31 @@ use crate::utils::sentry::ConfigureScope;
 
 use super::ResponseError;
 
+struct MultipartField<'a> {
+    inner: axum::extract::multipart::Field<'a>,
+}
+
+impl<'a> MultipartField<'a> {
+    fn new(inner: axum::extract::multipart::Field<'a>) -> Self {
+        Self { inner }
+    }
+}
+
+impl<'a> Stream for MultipartField<'a> {
+    type Item = Result<bytes::Bytes, std::io::Error>;
+
+    fn poll_next(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let this = &mut self;
+        let inner = std::pin::Pin::new(&mut this.inner);
+        inner
+            .poll_next(cx)
+            .map_err(|_err| std::io::Error::new(std::io::ErrorKind::Other, "oops"))
```

Comment:
I think this here is the critical part, right? tokio expects everthing to be `io::Error`, and while the `Field` itself implements `Stream`, it has a different error type? Why doesn’t `StreamReader` offer a `map_err` combinator, that would be so convenient.

Reply:
yes, i just added some comments to clarify this.  And yeah, having to newtype this just to change the error type is a pain :(